### ISSUE TITLE
HTML+Angular formatting generates invalid HTML

### ISSUE THREAD

**Issue by ericsnap** [prettier/prettier#7677]
**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIwA2AhgM4kC8AOuEQE7q4C2ARgLQDMuxAnnLXOlaZ4jagD5KUXNNwYc+YmSohaEAO64oEVgHMArjHi0S4yTPOyseQqQrVIBVo0HtquANpQdAZTXCwABYAusrCcIwAdJCwCDBR0PCwACrcAA5wplIW2aiYjDpm2UXSAFRevv4BAMKkcMoA5ACqJHwACgEQMBD11IXF2e4ktGAh1GGMAPzxMbBTei20ACJEMERTqR1dUzABeixQRJgEjbQEvVn9FjZK9nSCeQUgfZcy7kQEMKMg41PRiTBzBbLVbrTYQba7faHY6nc4vXAAegkF36cmwz3hZR8fhggRqLQa3ggBEw6CqxIgtAAQpT0Hwek8UfCSDBuAQ6tRmLS+EhcABGfiMXAkYmkgDccPhMhZbI5IACcEwOgCMF5fIATIKJYypa8ZeyIlz6HxWA5KV9xtN-hF5nxgURDdzaOSCJTJZcxKgEVZkfC0RiXliKrjqrVCaKyRTaAyA5drnYQCQAkR0OpcKo9FA6QxGBx3VL9XKFUqVWrNeFtbH+oNWQbmEQwABrHQZrMWkRW2I2oErB0AMyOZx1L093pwvpeaNwQZxgUWcD7RD0HwUtmUSZTadb2aYHFwqTzIAnutQJFSRCgx91MlQi9YPD4JA8A4AHgIAOqknZfGC0PQZEBhUwAAvOUAA4X0ya9UTvTBolwODoFCRgiB0ADcF-C8SD7SlGHXAJaEwKBG1YAA2Nx42UeAXxgJwDAEcQvVg6Ir2g8xbyIIR4MQqBlHrKAKMUBNqNoxh6PQRiEWY6BWOvJjOIfYxZMuL0zwvZSLC9H0qy08cdKsVcbnAYknEEABOKDJwMyjbnoVguXQbhLJPAIAFZDITMA7lYGBhHZcRgGABCO1ZdJcAAX3C3AAApAuC8JO1gRKYBSMLIoASi9NyNOKKcbJoOyROctiMAIDzlEEPt2RfXAqrgF9WDUWgiFSYq2JvbSmXazT0BgcrqBEuj4AYEgUIIIcxBfBFuC0mAcu6jB0DEOLLT+Q4oD4CIasixFcBWjs1qIzbuAi8KtKWqtoN07B5pPTrutRXr+pAQaxOG4UxomjQaoCWbbpK9Alv2hLDo22gIg0HaauByJQc2gJTvO-6-THG7Lrk+6HqKDA+vy17xI+94JtUVZfOgP70faxblqC1aEnWzaScR70Lq6qnUeR7HvQITnpGu3nZGYAwuikfLmBgKRxage8iMbNxosIODG3S-D1AAKRFKRyFwABCJN1c1gKgv1tQNegXAJlweoAlJOB6lwXl6hN+2drV7wAHkADkvSFwwZMpzTIH4U1iWIVIFg8Bww5adAvj1jpTcNo9UFSfhntTuBWBIMBVHGohmH85PIDpGn4qFAAfXAACtNeZ4u4FHDPRyDzOo5ahZ-v5nSOeeLuoD7kAABoQAgVIyagEhkFAOhVDUVo6AQSeUCIbAIFJIeQGYZqmzgGBvHPMAiJ0ZBf3-YeVUYAh3xt+A1LAOBvEX4QcGEbhkHAMgN6IhYYFaZqdBQsgRcBAWjDxri+Kk29Gy728EQRgcAAAyR0gHvFASAcB3gj7sgAIp6E6HAFBIC4DD3PMYPg78Lz6GILQDeqciIwE-L1AIyAwIAAYSGqBaO+ZqqR34ZwWNgAhw8ACOeD4B-1HkvEApBWAbQEAIDe-BRGYH4H-VCgCkDALQS0RgmAT5-mIYmLBcBcH4MIWg1YzBGE7GQOqYemEjhH3JIwDRIAtAbQ3raJI+cl5aMMdgf8ABJLMsRvA50wGPAAglmbwtYCGaNQXASKQA)
```sh
--parser angular
```

**Input:**
```html
<div class="card mb-3 layered-item">
    <div class="row no-gutters">
        <div class="col-md-3" [ngSwitch]="item.content.contentType">
            <img
                *ngSwitchCase="'UserPhoto'"
                [src]="item?.content?.userData?.photo?.thumbnailUrl"
                class="card-img"
                [alt]="item?.content?.userData?.photo?.thumbnailUrl"
            />
            <div
                *ngSwitchCase="'SolidColorBorder'"
                style="border: 1rem solid;"
                style="height: 12rem;"
                [style.border-color]="item.content.userData.borderColor"
            ></div>
            <div
                *ngSwitchCase="'SolidColor'"
                class="shadow rounded m-3"
                style="height: 12rem;"
                [style.background]="item.content.userData.fill"
            ></div>
            <div *ngSwitchDefault class="shadow rounded m-3 p-3">
                <span>
                    <fa-layers [fixedWidth]="true" size="8x">
                        <fa-icon icon="image" transform="shrink-6" class="text-muted"></fa-icon>
                        <fa-icon icon="ban" class="text-muted"></fa-icon>
                    </fa-layers>
                </span>
            </div>
        </div>
        <div class="col-md-9">
            <div class="card-body">
                <h5 class="card-title">{{ item.type }} ({{ item.content.contentType }})</h5>
                <div class="card-text">
                    <dl class="d-flex flex-wrap">
                        <div>
                            <dt class="text-muted small">x/y</dt>
                            <dd>{{ item.container.x }} / {{ item.container.y }}</dd>
                        </div>
                        <div>
                            <dt class="text-muted small">w x h</dt>
                            <dd>{{ item.container.w }} x {{ item.container.h }}</dd>
                        </div>
                        <div>
                            <dt class="text-muted small">rotation</dt>
                            <dd>{{ item.container.rot }}</dd>
                        </div>
                    </dl>
                </div>
                <button class="btn btn-link" (click)="showJson = !showJson">{{ showJson ? 'hide' : 'show' }} JSON</button>
                <core-collapser [collapsed]="!showJson"><pre class="pre-scrollable"><code>{{ item | json }}</code></pre></core-collapser>
            </div>
        </div>
    </div>
</div>

```

**Output:**
```html
<div class="card mb-3 layered-item">
  <div class="row no-gutters">
    <div class="col-md-3" [ngSwitch]="item.content.contentType">
      <img
        *ngSwitchCase="'UserPhoto'"
        [src]="item?.content?.userData?.photo?.thumbnailUrl"
        class="card-img"
        [alt]="item?.content?.userData?.photo?.thumbnailUrl"
      />
      <div
        *ngSwitchCase="'SolidColorBorder'"
        style="border: 1rem solid;"
        style="height: 12rem;"
        [style.border-color]="item.content.userData.borderColor"
      ></div>
      <div
        *ngSwitchCase="'SolidColor'"
        class="shadow rounded m-3"
        style="height: 12rem;"
        [style.background]="item.content.userData.fill"
      ></div>
      <div *ngSwitchDefault class="shadow rounded m-3 p-3">
        <span>
          <fa-layers [fixedWidth]="true" size="8x">
            <fa-icon
              icon="image"
              transform="shrink-6"
              class="text-muted"
            ></fa-icon>
            <fa-icon icon="ban" class="text-muted"></fa-icon>
          </fa-layers>
        </span>
      </div>
    </div>
    <div class="col-md-9">
      <div class="card-body">
        <h5 class="card-title">
          {{ item.type }} ({{ item.content.contentType }})
        </h5>
        <div class="card-text">
          <dl class="d-flex flex-wrap">
            <div>
              <dt class="text-muted small">x/y</dt>
              <dd>{{ item.container.x }} / {{ item.container.y }}</dd>
            </div>
            <div>
              <dt class="text-muted small">w x h</dt>
              <dd>{{ item.container.w }} x {{ item.container.h }}</dd>
            </div>
            <div>
              <dt class="text-muted small">rotation</dt>
              <dd>{{ item.container.rot }}</dd>
            </div>
          </dl>
        </div>
        <button class="btn btn-link" (click)="showJson = !showJson">
          {{ showJson ? "hide" : "show" }} JSON
        </button>
        <core-collapser [collapsed]="!showJson"
          >><pre
            class="pre-scrollable"
          ><code>{{ item | json }}</code></pre></core-collapser</core-collapser
        >
      </div>
    </div>
  </div>
</div>

```

**Expected behavior:**

Around line 60 of the output, it generates an incorrect `>><pre` (double `>`). Then, 2 lines below, it seems to have trouble closing the tag `</core-collapser</core-collapser` (the closing `>` is on the next line - which is not an issue).

Angular does not complain about the unformatted version of the file, so I don't see any reason to believe it has a syntax error to start with.

Any `core-*` is a component from a shared library we wrote. Any `app-*` is obviously a local component to the project.

**Comment by thorn0** [comment]
Seems to have been fixed by #5959, but not yet released. Please confirm that the formatting is okay on the playground of that PR.

**Prettier pr-5959**
[Playground link](https://deploy-preview-5959--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIwA2AhgM4kC8AOuEQE7q4C2ARgLQDMuxAnnLXOlaZ4jagD5KUXNNwYc+YmSohaEAO64oEVgHMArjHi0S4yTPOyseQqQrVIBVo0HtquANpQdAZTXCwABYAusrCcIwAdJCwCDBR0PCwACrcAA5wplIW2aiYjDpm2UXSAFRevv4BAMKkcMoA5ACqJHwACgEQMBD11IXF2e4ktGAh1GGMAPzxMbBTei20ACJEMERTqR1dUzABeixQRJgEjbQEvVn9FjZK9nSCeQUgfZcy7kQEMKMg41PRiTBzBbLVbrTYQba7faHY6nc4vXAAegkF36cmwz3hZR8fhggRqLQa3ggBEw6CqxIgtAAQpT0Hwek8UfCSDBuAQ6tRmLS+EhcABGfiMXAkYmkgDccPhMhZbI5IACcEwOgCMF5fIATIKJYypa8ZeyIlz6HxWA5KV9xtN-hF5nxgURDdzaOSCJTJZcxKgEVZkfC0RiXliKrjqrVCaKyRTaAyA5drnYQCQAkR0OpcKo9FA6QxGBx3VL9XKFUqVWrNeFtbH+oNWQbmEQwABrHQZrMWkRW2I2oErB0AMyOZx1L093pwvpeaNwQZxgUWcD7RD0HwUtmUSZTadb2aYHFwqTzIAnutQJFSRCgx91MlQi9YPD4JA8A4AHgIAOqknZfGC0PQZEBhUwAAvOUAA4X0ya9UTvTBolwODoFCRgiB0ADcF-C8SD7SlGHXAJaEwKBG1YAA2Nx42UeAXxgJwDAEcQvVg6Ir2g8xbyIIR4MQqBlHrKAKMUBNqNoxh6PQRiEWY6BWOvJjOIfYxZMuL0zwvZSLC9H0qy08cdKsVcbnAYknEEABOKDJwMyjbnoVguXQbhLJPAIAFZDITMA7lYGBhHZcRgGABCO1ZdJcAAX3C3AAApAuC8JO1gRKYBSMLIoASi9NyNOKKcbJoOyROctiMAIDzlEEPt2RfXAqrgF9WDUWgiFSYq2JvbSmXazT0BgcrqBEuj4AYEgUIIIcxBfBFuC0mAcu6jB0DEOLLT+Q4oD4CIasixFcBWjs1qIzbuAi8KtKWqtoN07B5pPTrutRXr+pAQaxOG4UxomjQaoCWbbpK9Alv2hLDo22gIg0HaauByJQc2gJTvO-6-THG7Lrk+6HqKDA+vy17xI+94JtUVZfOgP70faxblqC1aEnWzaScR70Lq6qnUeR7HvQITnpGu3nZGYAwuikfLmBgKRxage8iMbNxosIODG3S-D1AAKRFKRyFwABCJN1c1gKgv1tQNegXAJlweoAlJOB6lwXl6hN+2drV7wAHkADkvSFwwZMpzTIH4U1iWIVIFg8Bww5adAvj1jpTcNo9UFSfhntTuBWBIMBVHGohmH85PIDpGn4qFAAfXAACtNeZ4u4FHDPRyDzOo5ahZ-v5nSOeeLuoH5kAABoQAgVIyagEhkFAOhVDUVo6AQSeUCIbAIFJIeQGYZqmzgGBvHPMAiJ0ZBf3-YeVUYAh3xt+A1LAOBvEX4QcGEbhkHAMgN6IhYYFaZqdBQsgRcBAWjDxri+Kk29Gy728EQRgcAAAyR0gHvFASAcB3gj7sgAIp6E6HAFBIC4DD3PMYPg78Lz6GILQDeqciIwE-L1AIyAwIAAYSGqBaO+ZqqR34ZwWNgAhw8ACOeD4B-1HkvEApBWAbQEAIDe-BRGYH4H-VCgCkDALQS0RgmAT5-mIYmLBcBcH4MIWg1YzBGE7GQOqYemEjhH3JIwDRIAtAbQ3raJI+cl5aMMdgf8ABJLMsRvA50wGPAAglmbwtYCGaNQXASKQA)
```sh
--parser angular
```

**Input:**
```html
<div class="card mb-3 layered-item">
    <div class="row no-gutters">
        <div class="col-md-3" [ngSwitch]="item.content.contentType">
            <img
                *ngSwitchCase="'UserPhoto'"
                [src]="item?.content?.userData?.photo?.thumbnailUrl"
                class="card-img"
                [alt]="item?.content?.userData?.photo?.thumbnailUrl"
            />
            <div
                *ngSwitchCase="'SolidColorBorder'"
                style="border: 1rem solid;"
                style="height: 12rem;"
                [style.border-color]="item.content.userData.borderColor"
            ></div>
            <div
                *ngSwitchCase="'SolidColor'"
                class="shadow rounded m-3"
                style="height: 12rem;"
                [style.background]="item.content.userData.fill"
            ></div>
            <div *ngSwitchDefault class="shadow rounded m-3 p-3">
                <span>
                    <fa-layers [fixedWidth]="true" size="8x">
                        <fa-icon icon="image" transform="shrink-6" class="text-muted"></fa-icon>
                        <fa-icon icon="ban" class="text-muted"></fa-icon>
                    </fa-layers>
                </span>
            </div>
        </div>
        <div class="col-md-9">
            <div class="card-body">
                <h5 class="card-title">{{ item.type }} ({{ item.content.contentType }})</h5>
                <div class="card-text">
                    <dl class="d-flex flex-wrap">
                        <div>
                            <dt class="text-muted small">x/y</dt>
                            <dd>{{ item.container.x }} / {{ item.container.y }}</dd>
                        </div>
                        <div>
                            <dt class="text-muted small">w x h</dt>
                            <dd>{{ item.container.w }} x {{ item.container.h }}</dd>
                        </div>
                        <div>
                            <dt class="text-muted small">rotation</dt>
                            <dd>{{ item.container.rot }}</dd>
                        </div>
                    </dl>
                </div>
                <button class="btn btn-link" (click)="showJson = !showJson">{{ showJson ? 'hide' : 'show' }} JSON</button>
                <core-collapser [collapsed]="!showJson"><pre class="pre-scrollable"><code>{{ item | json }}</code></pre></core-collapser>
            </div>
        </div>
    </div>
</div>
```

**Output:**
```html
<div class="card mb-3 layered-item">
  <div class="row no-gutters">
    <div class="col-md-3" [ngSwitch]="item.content.contentType">
      <img
        *ngSwitchCase="'UserPhoto'"
        [src]="item?.content?.userData?.photo?.thumbnailUrl"
        class="card-img"
        [alt]="item?.content?.userData?.photo?.thumbnailUrl"
      />
      <div
        *ngSwitchCase="'SolidColorBorder'"
        style="border: 1rem solid;"
        style="height: 12rem;"
        [style.border-color]="item.content.userData.borderColor"
      ></div>
      <div
        *ngSwitchCase="'SolidColor'"
        class="shadow rounded m-3"
        style="height: 12rem;"
        [style.background]="item.content.userData.fill"
      ></div>
      <div *ngSwitchDefault class="shadow rounded m-3 p-3">
        <span>
          <fa-layers [fixedWidth]="true" size="8x">
            <fa-icon
              icon="image"
              transform="shrink-6"
              class="text-muted"
            ></fa-icon>
            <fa-icon icon="ban" class="text-muted"></fa-icon>
          </fa-layers>
        </span>
      </div>
    </div>
    <div class="col-md-9">
      <div class="card-body">
        <h5 class="card-title">
          {{ item.type }} ({{ item.content.contentType }})
        </h5>
        <div class="card-text">
          <dl class="d-flex flex-wrap">
            <div>
              <dt class="text-muted small">x/y</dt>
              <dd>{{ item.container.x }} / {{ item.container.y }}</dd>
            </div>
            <div>
              <dt class="text-muted small">w x h</dt>
              <dd>{{ item.container.w }} x {{ item.container.h }}</dd>
            </div>
            <div>
              <dt class="text-muted small">rotation</dt>
              <dd>{{ item.container.rot }}</dd>
            </div>
          </dl>
        </div>
        <button class="btn btn-link" (click)="showJson = !showJson">
          {{ showJson ? "hide" : "show" }} JSON
        </button>
        <core-collapser [collapsed]="!showJson">
          <pre class="pre-scrollable"><code>{{ item | json }}</code></pre>
        </core-collapser>
      </div>
    </div>
  </div>
</div>

```

**Comment by ericsnap** [comment]
@thorn0 you are correct. I ran it in there and the formatting is correct. I will close this issue then and wait for the release.
